.github/workflows/release_sdist.yml

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+    name: sdistRelease
+    
+    on:  # Specifies the event triggering the workflow
+      workflow_call:  # Indicates that this is a reusable workflow
+        inputs:
+          os:
+            required: true
+            type: string
+    
+    
+    # Use MACOSX_DEPLOYMENT_TARGET=12.0 to produce compatible wheel
+    env:
+      MACOSX_DEPLOYMENT_TARGET: "12.0"
+    
+    permissions:
+      contents: read
+    
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name == 'workflow_dispatch' }}-sourcedist
+    
+    jobs:
+      build:
+        if: github.event_name != 'pull_request' || startsWith( github.base_ref, 'rel-') || contains( github.event.pull_request.labels.*.name, 'run release CIs') 
+        runs-on: macos-latest
+        strategy:
+          matrix:
+            python-version: ['3.10']
+            target-architecture: ['arm64']
+    
+        steps:
+        - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    
+    
+        - name: Checkout submodules
+          shell: bash
+          run: |
+            auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+            git submodule sync --recursive
+            git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+            
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+          with:
+            python-version: ${{ matrix.python-version }}
+    
+        - name: Install Python dependencies
+          run: |
+            python -m pip install -q --upgrade pip
+            python -m pip install -q -r requirements-release.txt
+    
+        # Only triggered by weekly event on certain CI
+        - name: Build source distribution (weekly)
+          if: github.event_name == 'schedule'  && matrix.python-version == '3.10' && matrix.target-architecture == 'arm64'
+          run: |
+            # Build and upload source distribution to PyPI
+            git clean -xdf
+            sed -i 's/name = "onnx"/name = "onnx-weekly"/' 'pyproject.toml'
+            ONNX_PREVIEW_BUILD=1 python -m build --sdist
+    
+        - name: Build source distribution (for release)
+          if: matrix.python-version == '3.10' && matrix.target-architecture == 'arm64'
```

Comment:
Is this safe?

Reply:
What exactly do you mean by that? It is at least the current behavior?